Results_Engine/Query/Max.cs

```diff
@@ -50,21 +55,62 @@ public static List<BarForce> MaxForces(this IEnumerable<BarForce> forces)
+        {
+            List<BarRequiredArea> maxResults = new List<BarRequiredArea>();
+            maxResults.Add(results.MaxBy(x => x.Top));
+            maxResults.Add(results.MaxBy(x => x.Bottom));
+            maxResults.Add(results.MaxBy(x => x.Perimeter));
+            maxResults.Add(results.MaxBy(x => x.Shear));
+            maxResults.Add(results.MaxBy(x => x.Torsion));
+
+            return maxResults;
+        }
+
+        /***************************************************/
+        [Description("Determines the maximum required area for each component of the MeshRequiredArea and returns a new MeshRequiredArea with its concurrent required areas.")]
+        [Input("results", "The MeshRequiredArea results to be considered.")]
+        [Output("maxResults", "A list of MeshRequiredAreas, one for each component of the MeshRequiredAreas, containing the maximum required area and its concurrent required areas.")]
+        public static List<MeshRequiredArea> MaxMeshRequiredArea(this IEnumerable<MeshRequiredArea> results)
+        {
+            List<MeshRequiredArea> maxResults = new List<MeshRequiredArea>();
+            maxResults.Add(results.MaxBy(x => x.TopPrimary));
+            maxResults.Add(results.MaxBy(x => x.TopSecondary));
+            maxResults.Add(results.MaxBy(x => x.BottomPrimary));
+            maxResults.Add(results.MaxBy(x => x.BottomSecondary));
+            maxResults.Add(results.MaxBy(x => x.Shear));
+            maxResults.Add(results.MaxBy(x => x.Torsion));
+
+            return maxResults;
+        }
 
+        /***************************************************/
+        [Description("Groups the BarForces by case and finds the maximum envelope returning a single enveloped BarForce for each case.")]
+        [Input("forces", "The BarForces to be considered.")]
+        [Output("maxResults", "A list of BarForces, one for each case, enveloped to produce the maximum forces for that case.")]
         public static List<BarForce> MaxEnvelopeByCase(this IEnumerable<BarForce> forces)
         {
             return forces.GroupByCase().Select(x => x.MaxEnvelope(false, true)).ToList();
         }
 
         /***************************************************/
-
+        [Description("Groups the BarForces by id and finds the maximum envelope returning a single enveloped BarForce for each id.")]
+        [Input("forces", "The BarForces to be considered.")]
+        [Output("maxResults", "A list of BarForces, one for each id, enveloped to produce the maximum forces for that id.")]
         public static List<BarForce> MaxEnvelopeByObject(this IEnumerable<BarForce> forces)
         {
             return forces.GroupByObjectId().Select(x => x.MaxEnvelope(true, false)).ToList();
         }
 
         /***************************************************/
-
+        [Description("Determines the maximum force in each component of the BarForce and returns a single BarForce object with the enveloped forces.")]
```

Comment:
Would note here that the resulting force will not be concurrent. Same for all methods where this is true.